data-science-onramp/ai-platform/modules/trainer/tfkeras_model/model.py

```diff
@@ -0,0 +1,91 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Defines a Keras model and input function for training."""
+
+# [START ai_platform_tf_keras_model]
+# [START ai_platform_tfkeras_model_tf_import]
+import tensorflow as tf
+# [END ai_platform_tfkeras_model_tf_import]
+
+# [START ai_platform_tfkeras_model_input_fn]
+def input_fn(features, labels, num_epochs, batch_size, shuffle):
+    """Generates an input function to be used for model training"""
+
+    inputs = (features, labels)
+    dataset = tf.data.Dataset.from_tensor_slices(inputs)
+
+    if shuffle:
+        dataset = dataset.shuffle(buffer_size=len(features))
+
+    # Call repeat after shuffling to prevent epochs from blending together
+    dataset = dataset.repeat(num_epochs)
+    dataset = dataset.batch(batch_size)
+    return dataset
+# [END ai_platform_tfkeras_model_input_fn]
+
+# [START ai_platform_tfkeras_model_create_keras_model]
+# [START ai_platform_tfkeras_model_create_keras_model_init]
+def create_keras_model(input_dim, output_dim, learning_rate):
+    """Creates Keras Model for regression"""
+
+    # Define regularizers
+    k_regularizer = tf.keras.regularizers.l1_l2(l1=1e-5, l2=1e-4)
+    b_regularizer = tf.keras.regularizers.l2(1e-4)
+# [END ai_platform_tfkeras_model_create_keras_model_init]
+# [START ai_platform_tfkeras_model_create_keras_model_define]
+    # Define model layers
+    Dense = tf.keras.layers.Dense
+    # Define Keras model
+    model = tf.keras.Sequential(
+        [
+            Dense(
+                11,
+                activation=tf.nn.relu,
+                input_shape=(input_dim,),
+                kernel_regularizer=k_regularizer,
+                bias_regularizer=b_regularizer,
+            ),
+            Dense(80, activation=tf.nn.relu),
+            Dense(150, activation=tf.nn.relu),
+            Dense(
+                300,
+                activation=tf.nn.relu,
+                kernel_regularizer=k_regularizer,
+                bias_regularizer=b_regularizer,
+            ),
+            Dense(500, activation=tf.nn.relu),
+            Dense(
+                800,
+                activation=tf.nn.relu,
+                kernel_regularizer=k_regularizer,
+                bias_regularizer=b_regularizer,
+            ),
+            Dense(1000, activation=tf.nn.relu),
+            Dense(
+                1500,
+                activation=tf.nn.relu,
+                kernel_regularizer=k_regularizer,
+                bias_regularizer=b_regularizer,
+            ),
+            Dense(output_dim),
+        ]
+    )
```

Comment:
@ivanmkc what do you think?

Reply:
Sorry just seeing this. I don't know if this tutorial's scope covers the internals of the custom model. If not, then it might be okay to just note that this is just one example of a model definition, written using the Keras API.